src/main/groovy/com/cloudcard/photoDownloader/DatabaseStorageService.java

```diff
@@ -58,16 +58,15 @@ void init() {
 
 
     @Override
-    public List<PhotoFile> save(Collection<Photo> photos) {
-
+    public StorageResults save(Collection<Photo> photos) {
         List<PhotoFile> photoFiles = new ArrayList<>();
-
+        //TODO: Implement returning failed photos to this loop
```

Comment:
rephrase - TODO consider if persistent failures are possible with this storage service, and return UnsaveablePhotoFile instances in those cases.